frontend/src/components/RegionPermissionModal.js

```diff
@@ -0,0 +1,100 @@
+import React, { useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import Modal from './Modal';
+import { DECIMAL_BASE } from '../Constants';
+import { getUserRegions } from '../permissions';
+import './RegionPermissionModal.css';
+
+function RegionPermissionModal({
+  filters, user, showFilterWithMyRegions,
+}) {
+  const modalRef = useRef();
+  const userRegions = getUserRegions(user);
+
+  useEffect(() => {
+    // Check if user has permission to region.
+    const regionFilters = filters.filter((f) => f.topic === 'region');
+    if (regionFilters.length > 0) {
+      let showRegionPermissionsModal = false;
+      const deniedRegionsList = [];
+      regionFilters.forEach((f) => {
+        const filterRegion = f.query;
+        const filterRegionNum = parseInt(filterRegion, DECIMAL_BASE);
+        if (!userRegions.includes(filterRegionNum)) {
+          showRegionPermissionsModal = true;
+          deniedRegionsList.push(filterRegionNum);
+        }
+      });
+
+      if (showRegionPermissionsModal && !modalRef.current.modalIsOpen) {
+        // Show region permission modal.
+        modalRef.current.toggleModal(true);
+      }
+    }
+  }, [userRegions, filters]);
+
+  const showFilterWithMyRegionAccess = () => {
+    showFilterWithMyRegions();
+    modalRef.current.toggleModal(false);
+  };
+
+  const smartSheetAccessLink = 'https://app.smartsheetgov.com/b/form/f0b4725683f04f349a939bd2e3f5425a';
+  const requestSmartSheetAccess = () => {
+    // Open link in new tab and close modal.
+    window.open(smartSheetAccessLink, '_blank');
```

Comment:
[Avoid resorting to window.open()](https://developer.mozilla.org/en-US/docs/Web/API/Window/open#avoid_resorting_to_window.open)

Reply:
This has been changed at the modal level to take a button. We now pass a link to open the smart sheet page.